File: busca_binaria/busca_binaria.ts
function buscaBinaria(array:number [], item: number): number | undefined {
    let menor = 0
    let maior = array.length - 1

    while(menor <= maior){
        let meioIndex = Math.floor((menor + maior) / 2)
        let meio = array[meioIndex]

        if (meio == item ) return meioIndex

        if (meioIndex > item){
            maior = meioIndex - 1
        } else {
            menor = meioIndex + 1
        }
    }

    return undefined
}

const list:number[] = []

for (let i = 0; i < 10000; i++) {
    list.push(i)
}

var result = buscaBinaria(list, 900)

result === undefined 
    ? console.log("Número não encontrado") 
    : console.log(list[result])

File: busca_binaria/busca_binaria.py
def busca_binaria(item):
    lista_ordenada = gerar_lista()
    
    menor = 0
    maior = len(lista_ordenada)

    while menor <= maior:
        meio_index = round((menor + maior) / 2)
        meio = lista_ordenada[meio_index]

        if meio == item:
            return meio_index
        
        if (meio_index > item):
            maior = meio_index -1
        else:
            menor = meio_index + 1

    return None


def gerar_lista():
    lista_ordenada = []

    for i in range(1000):
        lista_ordenada.append(i)

    return lista_ordenada


print(busca_binaria(100))
